Source/WebCore/platform/graphics/mac/controls/SwitchMacUtilities.mm

```diff
@@ -82,6 +82,26 @@ static float easeInOut(const float progress)
     return -2.0f * pow(progress, 3.0f) + 3.0f * pow(progress, 2.0f);
 }
 
+static FloatRect trackRectForBounds(bool isVertical, const FloatRect& bounds, const FloatSize& size)
+{
+    auto heightDifference = isVertical ? bounds.height() - size.width() : bounds.height() - size.height();
+    auto trackYOffset = std::max((heightDifference / 2.0f), 0.0f);
+    return FloatRect { FloatPoint { bounds.x(), bounds.y() + trackYOffset }, size };
+}
+
+static float verticalInflatedRectHeightOffset(const IntOutsets& outsets, const ControlStyle& style)
+{
+    return outsets.top() * style.zoomFactor - outsets.bottom() * style.zoomFactor;
+}
```

Comment:
This might be unnecessary if you transpose outsets. See the vertical writing mode logic in `ProgressBarMac::rectForBounds`.

Reply:
I don't understand how you can adjust the outsets before you draw. Don't they have to be correct for horizontal drawing?

Reply:
The outsets simply determine the inflated rect, which is always done before drawing. There wouldn't be any change to horizontal mode, since the outsets would only be adjusted for vertical writing mode. 

That's what `ProgressBarMac` does in `rectForBounds` under `if (isVerticalWritingMode)`. 

Currently, the switch drawing logic calls `inflatedRect` directly. For example, `auto inflatedTrackRect = inflatedRect(trackRect, size, outsets, style);`. But I think it's time to implement `rectForBounds` to return an inflated rect that you can use to avoid additional adjustments later on.

Then, once you have that set up, similar to `ProgressBarMac::draw`, again under `if (isVerticalWritingMode)`, you'd transpose the the size of the inflated rect.